test/time/time_conversions.cpp

```diff
@@ -0,0 +1,102 @@
+/*=============================================================================
+Copyright 2020 Syed Ali Hasan <alihasan9922@gmail.com>
+
+Distributed under the Boost Software License, Version 1.0. (See accompanying
+file License.txt or copy at https://www.boost.org/LICENSE_1_0.txt)
+=============================================================================*/
+
+#define BOOST_TEST_MODULE time_conversions_test
+
+#include <iostream>
+#include <boost/units/systems/angle/degrees.hpp>
+#include <boost/astronomy/time/time_conversions.hpp>
+#include <boost/astronomy/time/parser.hpp>
+#include <boost/test/floating_point_comparison.hpp>
+
+#include <boost/test/unit_test.hpp>
+
+namespace tt = boost::test_tools;
+
+BOOST_AUTO_TEST_SUITE(time_conversions)
+
+BOOST_AUTO_TEST_CASE(time)
+{
+  //Make an object of universal time
+  //What was the GST at 14h 36m 51.67s UT on Greenwich date 22 April 1980?
+  //GST = 4h 40m 5.23s
+  //Verified from Practical Astronomy with your Calculator
+  //by Peter Duffett-Smith
+
+  std::string ts1("1980-04-22 14:36:51.67");
+  ptime t1(time_from_string(ts1));
+
+  DecimalHours d1 = GST(t1);
+
+  BOOST_CHECK_CLOSE(d1.get_hours(), 4, 0.5);
```

Comment:
Fix tests accuracy

Reply:
Regarding precission of minutes when calculating GST in time functionality
Expected Output: 4h 40m 5.23s
Output: 4h 42.m 2.00s

In every example minutes differ by 2minutes.

The algorithm I am using is accurate to 1/10th of a second.

The only problem is that calculating Julian Date is done by Boost which is not accurate.
**Expected Julian Day Number: 2444351.50
Julian Day Number Boost: 2444352.00**

This small difference of 0.5 is a magnifying error in GST

**Example 1.**

What was the GST at 19h 21m 0.0s UT on Greenwich date 10 April 1987?

JD Expected: 2446895.5
JD Boost: 2446896.00

Expected GST: 8h 34m 57.0896s = 8.5825hours
GST when JD Expected: 8.000000h 34.000000m 57.000000s = 8.5825hours
GST when using Boost JD: 8.000000h 36.000000m 55.000000s = 8.615278hours

Percentage Error: 0.3819%

**Example 2.**

What was the GST at 06h 02m 0.0s UT on Greenwich date 17 July 2020?

JD Expected: 2459047.5
JD Boost: 2459048.00

Expected GST: 01h 44m 06.667s = 1.735hours
GST when JD Expected: 1.000000h 44.000000m 6.000000s = 1.735hours
GST when using Boost JD:1.000000h 46.000000m 4.000000s = 1.767778hours

Percentage Error: 1.8892%

Reply:
Inexactness in julian_day() function. 
Percentage error set to 1% and proceeding.
https://github.com/boostorg/date_time/issues/167